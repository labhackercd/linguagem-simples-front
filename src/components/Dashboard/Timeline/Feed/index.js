import React from 'react';
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import {TwitterTweetEmbed} from 'react-twitter-embed';
import MoreOptionsIcon from './../../../../assets/more_options_icon.svg';
import { parseHourMinute } from './../../../Util';
import { ReactTinyLink } from 'react-tiny-link';

const useStyles = makeStyles((theme) => ({
  updatesArea: {
    overflow: 'auto',
    maxHeight: '50vh',
  },
  itemHeader: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  title: {
    color: theme.palette.primary.main
  },
  timelinePost: {
    overflow: 'auto',
    overflowWrap: 'break-word',
    display: 'flex',
    margin: '1rem 0',
    maxHeight: '25vh',
  },
}))

export default function Feed(props) {
  const classes = useStyles();
  return (
    <Grid container className={classes.updatesArea}>
      <List style={{width: '100%'}}>
          {props.updates ? props.updates.slice(0).reverse().map(update =>  //creates a shallow copy of the array and reverses it
						<ListItem divider key={update.id}>
              <Grid container>
                <Grid item xs={2}>
                  <Typography variant="h6">
										{parseHourMinute(update.created)}
                  </Typography>
                </Grid>
                <Grid item xs={10}>
                  <Grid container className={classes.itemHeader}>
                    <Typography variant="h5" className={classes.title}>
                      {update.title}
                    </Typography>
                    <Grid item>
                      <img src={MoreOptionsIcon} alt="more options icon"/>
                    </Grid>
                  </Grid>
                  <Grid container className={classes.timelinePost}>
                    <div style={{width: '100%'}}>
                      <Typography style={{color: '#666', alignSelf: 'flex-start'}} variant="body1">
												{update.updateTextArea ?
												 update.updateTextArea :
												 ''
												}</Typography>
                    </div>
                    <section style={{width: '100%'}}>
                      {update.tweet_id ?
                        <TwitterTweetEmbed
                          style={{width: '100%', margin: '0 0 1rem 0'}}
                          tweetId={update.tweet_id}
                        /> : '' }
                      {update.image ?
                       <img src={update.image}  style={{maxWidth: '100%', maxHeight: '30vh'}} alt="timeline post"/> : '' }
											{update.customURL && !(props.URLInputIsTwitter) ?
												<ReactTinyLink
													cardSize="small"
													showGraphic={true}
													maxLine={2}
													minLine={1}
													url={update.customURL} /> :
													''}
                    </section>
                  </Grid>
                </Grid>
              </Grid>
            </ListItem>
          ) : ''}
      </List>
    </Grid>
  )
}
