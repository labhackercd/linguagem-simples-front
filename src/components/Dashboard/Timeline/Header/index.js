import React, {useState} from 'react';
import {Button, Grid, Typography} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import InitStreamIcon from './../../../../assets/init_stream_button_icon.svg';
import StartBroadcastAlert from './../Dialogs/Alert/StartBroadcast';
import EndBroadcastAlert from './../Dialogs/Alert/EndBroadcast';

const useStyles = makeStyles((theme) => ({
  body: {
    width: '100%',
    margin: '1rem 0 -2rem 0rem',
  },
  titleRow: {
  	display: 'flex',
  	justifyContent: 'space-between',
  },
	title: {
		whiteSpace: 'nowrap',
		textDecoration: 'none',
		color: theme.palette.primary.main,
    justifyContent: 'flex-start',
	},
  buttonContainer: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  buttonBroadcastingOnline: {
    alignSelf: 'flex-start',
  	height: '30%',
  	color: '#FFF',
  	backgroundColor: '#00AF82',
  },
  buttonBroadcastingOffline: {
    alignSelf: 'flex-start',
    height: '30%',
    color: '#FFF',
    backgroundColor: '#000000',
  }
}));

export default function Header(props){
	const classes = useStyles();
  const [startBroadcastDialogOpen, setStartBroadcastDialogOpen] = useState(false)
  const [endBroadcastDialogOpen, setEndBroadcastDialogOpen] = useState(false)
  const [userInput, setUserInput] = useState(false)

  const handleStartBroadcastDialogOpen = (e) => {
    e.preventDefault()
    setStartBroadcastDialogOpen(true)
  }

  const handleEndBroadcastDialogOpen = (e) => {
    e.preventDefault()
    setEndBroadcastDialogOpen(true)
  }

  const handleDialogClose = (userInput) => {
    setStartBroadcastDialogOpen(false)
    setEndBroadcastDialogOpen(false)
    props.setBroadcastingStatus(userInput)
  }

	return (
    <div className={classes.body}>
  		<Grid container className={classes.titleRow}>
  			<Grid item md={6} >
          <Typography variant="h3" className={classes.title}>Linha do Tempo </Typography>
  			</Grid>
  			<Grid item md={6} className={classes.buttonContainer}>
          {props.broadcastingOnline ?
            <Button
              variant="contained"
              disableElevation
              id="end-stream"
              onClick={(e) => handleEndBroadcastDialogOpen(e)}
              className={classes.buttonBroadcastingOffline}
              startIcon={<img src={InitStreamIcon} alt="button to init stream"/>}>
              <h6>Finalizar transmissão</h6>
            </Button> :
            <Button
    	        variant="contained"
              id="start-stream"
    	        disableElevation
    	        className={classes.buttonBroadcastingOnline}
              onClick={(e) => handleStartBroadcastDialogOpen(e)}
    	        startIcon={<img src={InitStreamIcon} alt="button to init stream"/>}>
    	        <h6>Iniciar transmissão</h6>
    	      </Button>
          }
  			</Grid>
  		</Grid>
      <StartBroadcastAlert keepMounted
                      open={startBroadcastDialogOpen}
                      onClose={handleDialogClose}
                      value={userInput} />
      <EndBroadcastAlert keepMounted
                      open={endBroadcastDialogOpen}
                      onClose={handleDialogClose}
                      value={userInput} />
    </div>
	)
}
