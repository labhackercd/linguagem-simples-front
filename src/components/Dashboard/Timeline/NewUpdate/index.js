import React, {useState, useEffect} from 'react'
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import DividerIcon from './../../../../assets/divider.svg';
import PictureUploadIcon from './../../../../assets/picture_upload.svg';
import TwitterIcon from './../../../../assets/twitter_icon.svg';
import LinkIcon from './../../../../assets/link.svg'
import { parseHourMinute } from './../../../Util';

const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 1rem 0',
  },
  textArea: {
    height: '100%',
    margin: '0rem 0 0 0',
    width: '100%',
    border: '0',
  },
  textField: {
		margin: '0 1rem 0 0',
		height: '100%',
		width: '100%',
	},
  submenu: {
    display: 'flex',
  },
  subMenuItem: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    margin: '0 0.5rem 0 1rem',
  },
  button: {
    height: '70%',
    alignSelf: 'flex-end',
  },
  notchedOutline: {
    border: '2px solid #F2F2F2',
    borderWidthBottom: '0px',
    color: theme.palette.secondary,
    borderRadius: '0 0 0 0',
  },
  time: {
    color: theme.palette.primary.main
  },
}))
export default function NewUpdate(props) {
  const classes = useStyles();
	const dateAux = props.date ? parseHourMinute(props.date) : parseHourMinute(new Date())
	const [date, setDate] = useState(dateAux)

	useEffect(() => {
		setInterval(() => tick(), 1000)
	}, clearInterval(1000));

	const tick = () => {
		setDate(parseHourMinute(new Date()))
	}

  return (
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Nova atualização </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
      <Grid item md={12}>
          <Grid container >
              <form className={classes.textArea} noValidate autoComplete="off">
                <TextField
                  id="newUpdateTextField"
                  multiline
                  rows={4}
									disabled={!props.broadcastingOnline}
                  variant="outlined"
                  className={classes.textField}
                  bgcolor="white"
                  name = "updateText"
                  value={props.updateTextArea}
                  onChange = {props.handleChange}
                  InputProps={{
                    classes: {
                      notchedOutline: classes.notchedOutline
                    },
                  }}
                />
              </form>
          </Grid>
          <Box borderTop={1} color="#F2F2F2" borderRadius="0 0 10px 25px" bgcolor="#F2F2F2">
            <Grid container>
              <Grid item xs={8} className={classes.submenu}>
                <div className={classes.subMenuItem}>
                  <Typography className={classes.time} variant="h6"> {date} </Typography>
                </div>
                <div className={classes.subMenuItem}>
                  <img src={DividerIcon} alt="divider icon"/>
                </div>
                <div className={classes.subMenuItem}>
                  <a href="/"><img src={PictureUploadIcon} id={"picture-upload-icon"} alt="upload icon" onClick={(e) => props.handleDialogStateAction(e, true, "previewDialog", "InputImage")}/></a>
                </div>
								<div className={classes.subMenuItem}>
									<a href="/"><img src={TwitterIcon} id={"tweet-insert-icon"}alt="incorporate twitter icon" onClick={(e) => props.handleDialogStateAction(e, true, "URLInputDialog", "URLInputIsTwitter")}/></a>
								</div>
								<div className={classes.subMenuItem}>
									<a href="/"><img src={LinkIcon} id={"link-insert-icon"} alt="incorporate url icon" onClick={(e) => props.handleDialogStateAction(e, true, "URLInputDialog", "noAction")}/></a>
								</div>
              </Grid>
              <Grid item xs={4} style={{display: 'flex', justifyContent: 'flex-end'}}>
                <Button disabled={!props.broadcastingOnline} id={"updateSubmitButton"}className={classes.button} onClick={props.handleClick} variant="contained" disableElevation>
                  Inserir
                </Button>
              </Grid>
            </Grid>
          </Box>
      </Grid>
    </Grid>
  )
}
