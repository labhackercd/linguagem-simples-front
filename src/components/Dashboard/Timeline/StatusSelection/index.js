import React, {useState} from 'react';
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 1rem 0',
  },
  tabs: {
    display: 'flex',
    minHeight: '0px',
    margin: '0.5rem 0 0 0',
    overflow: 'auto',
    flexWrap: 'no-wrap',
  },
  tab: {
    whiteSpace: 'nowrap',
    padding: '0 0.5rem',
    margin: '0 0.3rem',
    backgroundColor: '#F2F2F2',
    borderRadius: '5px',
    minHeight: '0px',
    minWidth: 'auto',
    textTransform: 'capitalize'
  },
}))
export default function StatusSelection(props) {
  const classes = useStyles();
  const [titlesArray] = useState([
                                  'Pré-sessão',
                                  'Sessão Iniciada',
                                  'Votação Iniciada',
                                  'Votação Encerrada',
                                  'Próxima Pauta',
                                  'Sessão Encerrada',
                                  'Fim da transmissão'])
  return (
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader} style={{margin: '1rem 0 0 0'}}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Status </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
      <Grid className={classes.tabs}>
          {titlesArray.map(function(title, index){
          return <Button key={index} uvariant="contained"
												 onClick={(e) => props.startUpdateWithTitleFlow(e, title)}
												 disableElevation
												 disabled={!props.broadcastingOnline}
												 className={classes.tab}>{title}</Button>
        })}
      </Grid>
    </Grid>
  )
}
