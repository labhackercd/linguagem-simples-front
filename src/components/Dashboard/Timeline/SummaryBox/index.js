import React, {useState} from 'react';
import { Grid, Typography, Box, Button, TextField } from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import axiosInstance from '../../../../auth/axiosApi.js';
import DownArrowIcon from './../../../../assets/down-arrow.svg';
import {API_SESSIONS_URL} from './../../../../api_urls'

const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 0.5rem 0',
  },
	notchedOutline: {
		border: '2px solid #F2F2F2',
		borderWidthBottom: '0px',
		color: theme.palette.secondary,
		borderRadius: '5px 5px 0 0',
	},
	textArea: {
		height: '100%',
		margin: '0rem 0 0 0',
		width: '100%',
		border: '0',
	},
	button: {
	  height: 'inherit',
	  alignSelf: 'center',
		borderRadius: '0 0 5px 5px',
		backgroundColor: theme.palette.verdeCamaraLight.main,
		color: theme.palette.white.main,
		textTransform: 'capitalize',
	},
	textField: {
		margin: '0 1rem 0 0',
		height: '100%',
		width: '100%',
	},
	box: {
		borderRadius: '0 0 0 5px',
		width: '100%',
	},
	buttonContainer: {
		display: 'flex',
		justifyContent: 'flex-end',
		 height: 'inherit'
	},
	container: {
		height: '100%',
	},
}));

export default function SummaryBox(props) {
  const classes = useStyles();
  const sessionID = props.sessionID;
  const [textFieldValue, setTextFieldValue] = useState('');
  function handleChange(e) {
    setTextFieldValue(e.target.value)
  }
  function handleSubmit() {
    const url = API_SESSIONS_URL + sessionID + '/';
    axiosInstance.patch(url, {
            id: sessionID,
            resume: textFieldValue
        }).then(
            result => {
              if(result.status === 200) {
                alert('Resumo atualizado com sucesso!')
              }
            }
    ).catch (error => {
        console.log(error)
    })

  }
  return (
    <React.Fragment>
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Resumo </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
    </Grid>
    <Grid item md={12}>
        <Grid container >
            <form className={classes.textArea} noValidate autoComplete="off">
              <TextField
                id="summary-box-textfield"
                multiline
                rows={4}
                value={textFieldValue}
                onChange={handleChange}
                variant="outlined"
								disabled={!props.broadcastingOnline}
                className={classes.textField}
                bgcolor="white"
                InputProps={{
                  classes: {
                    notchedOutline: classes.notchedOutline
                  },
                }}
              />
            </form>
          <Box className={classes.box} color="#F2F2F2" bgcolor="#F2F2F2">
						<Grid container className={classes.container}>
								<Grid item xs={8}></Grid>
								<Grid item xs={4} className={classes.buttonContainer}>
									<Button disabled={!props.broadcastingOnline} id="summary-box-submit-button" className={classes.button} onClick={(e) => handleSubmit(e)} variant="contained" disableElevation>
										Atualizar
									</Button>
								</Grid>
						</Grid>
          </Box>
      </Grid>
		</Grid>
    </React.Fragment>
  )
}
