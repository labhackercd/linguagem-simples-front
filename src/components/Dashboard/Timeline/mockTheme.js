import React from 'react';
import { ThemeProvider } from '@material-ui/core/styles';
import { createMuiTheme } from '@material-ui/core';
export default function MockTheme({ children }: any) {
  const theme = createMuiTheme({
    typography: {
      fontFamily: "OpenSans",
      body:{
        fontSize:12,
        fontStyle: 'semibold',
      },
      h1:{
        fontSize:54,
        fontStyle: 'semibold',
      },
      h2:{
        fontSize:36,
        fontStyle: 'semibold',
      },
      h3:{
        fontSize:24,
        fontStyle: 'semibold',
      },
      h4:{
        fontSize:16,
        fontStyle: 'semibold',
      },
      h5:{
        fontSize:14,
        fontStyle: 'semibold',
      },
      h6:{
        fontSize:16,
        fontStyle: 'semibold',
      },
      h7:{
        fontSize:9,
        fontStyle: 'regular',
      },
      body1:{
        fontSize:12,
        fontStyle: 'regular'
      },
      body2:{
        fontSize:9,
        fontStyle: 'semibold',
      }
    },
    palette: {
      primary: {
        main: '#007E5A',
      },
      cinza1: {
        main: '#F2F2F2',
      },
      cinza2: {
        main: '#C4C4C4',
      },
      grey: {
        main: '#666666',
      },
      amareloCamara: {
        main: '#FAC915',
      },
      verdeCamaraLight: {
        main: '#00AF82',
      },
      white: {
        main: '#FFF',
      }
    }});
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
