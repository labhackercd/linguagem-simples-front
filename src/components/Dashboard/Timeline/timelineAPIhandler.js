import axiosInstance from './../../../auth/axiosApi'
import axios from 'axios'
import {API_PUBLICATIONS_URL} from './../../../api_urls'

export async function fetchFeedUpdates(sessionId){
    if(!sessionId) { return "Please provide a valid session Id"}
    const response = await axiosInstance.get((API_PUBLICATIONS_URL + '?session__id=' + sessionId))
    if(response.status === 200) {
      return response.data
    }
    return response
}
