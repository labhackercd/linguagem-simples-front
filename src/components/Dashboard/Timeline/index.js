import React, {useState} from 'react';
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import {TwitterTweetEmbed} from 'react-twitter-embed';
import SummaryBox from './SummaryBox';
import axiosInstance from '../../../auth/axiosApi.js';
import Header from './Header';
import StatusSelection from './StatusSelection';
import NewUpdate from './NewUpdate';
import TwitterDialog from './Dialogs/Twitter';
import ImageUploadDialog from './Dialogs/ImageUpload';
import PreviewDialog from './Dialogs/Preview';
import Feed from './Feed';

const useStyles = makeStyles((theme) => ({
	body: {
		padding: '0 1rem 0 1rem',
		height: 'calc(100vh-3px)',
	}
}));

export default function Timeline(props){
	const classes = useStyles();
	const sessionID = props.sessionID;
	const [updates, setUpdates] = useState([]);
	const [updateTitle, setUpdateTitle] = useState('');
	const [updateTextArea, setUpdateTextArea] = useState("");
	const [tweetURL, setTweetURL] = useState('');
	const [tweetID, setTweetID] = useState('');
	const [twitterDialogOpen, setTwitterDialogOpen] = useState(false);
	const [previewModalOpen, setPreviewModalOpen] = useState(false);
	const [imageUploadModalOpen, setImageUploadModalOpen] = useState(false);
	const [picture, setPicture] = useState(false);
	const [time] = useState('18:00');
	const errorMessages = {
				lacks_payload_content: 'Voce deve submeter pelo menos uma atualizacao, ou imagem ou tweet para prosseguir'
	}
	function dispatchPayload() {
		if (validatePayload()) {
			const formData = new FormData()
			if(picture) {
				formData.append('image', picture, picture.name)
			}
			formData.append('title', updateTitle)
			formData.append('content', updateTextArea)
			formData.append('session', sessionID)
			formData.append('tweet_id', tweetID)
			axiosInstance.post('/publications/', formData, {
				headers: { 'Content-Type': 'multipart/form-data'},
			}).then(result => {
				if(result.status===201) {
						let data = result.data
						console.log(data)
						let newUpdate = {
							id: data.id,
							content: data.content,
							time: parseHourMinute(new Date(data.created))
						}
						if(data.tweet_id.length > 0) {
							newUpdate['tweet_id'] = data.tweet_id
						}
						if(data.image){
							newUpdate['image'] = data.image
						}
						if(data.title) {
							newUpdate['title'] = data.title
						}
						setUpdates([...updates, newUpdate])
						garbageCollection()
				}
			}).catch(err => {
				console.log(err)
			})
		} else {
			alert(errorMessages.lacks_payload_content)
		}
	}
	{/* At least one field among content, image and tweet must not be null before we send a request to the API */}
	function validatePayload() {
		let contentExists = updateTextArea.length > 0
		let tweetExists = tweetID.length > 0
		return picture || contentExists || tweetExists;
	}
	function parseHourMinute(date) {
		return date.getHours() + ':' + ('0'+ date.getMinutes()).slice(-2)
	}
	function handleClick() {
		dispatchPayload()
	}
	function handleTwitterDialogOpen() {
		setTwitterDialogOpen(true)
	}
	function handleTwitterDialogClose() {
		setTwitterDialogOpen(false)
		let parseURL = tweetURL.split('/')
		let path = parseURL[parseURL.length-1]
		{/* removes suplemental information after the tweet's id */}
		let id = path.split('?')[0]
		setTweetID(id)
		setPreviewModalOpen(true)
	}
	function garbageCollection() {
		setTweetID('')
		setPicture(false)
		setUpdateTitle('')
		setUpdateTextArea('')
	}
	function handleImageUploadDialogOpen(e){
		e.preventDefault()
		setImageUploadModalOpen(true)
	}
	function handleImageUploadDialogClose() {
			setImageUploadModalOpen(false)
			dispatchPayload()
	}
	function handlePreviewModalClose() {
		dispatchPayload()
		setPreviewModalOpen(false)
	}
	function handleChange(e) {
		setUpdateTextArea(e.target.value)
	}
	function onImageDrop(picture) {
		picture.length > 0 ? setPicture(picture[0]) : setPicture(false)
	}
	function startUpdateWithTitleFlow(e, title) {
		handleSessionTitle(title)
		handleImageUploadDialogOpen(e)
	}
	function handleSessionTitle(title) {
		title.length > 0 ? setUpdateTitle(title) : setUpdateTitle('')
	}
	return (
		<div className={classes.body}>
			<Header></Header>
			<SummaryBox sessionID={sessionID}></SummaryBox>
			<StatusSelection startUpdateWithTitleFlow={startUpdateWithTitleFlow}></StatusSelection>
			<NewUpdate handleClick={handleClick}
								 handleTwitterDialogOpen={handleTwitterDialogOpen}
								 handleImageUploadDialogOpen={handleImageUploadDialogOpen}
								 updateTextArea={updateTextArea}
								 handleChange={handleChange}></NewUpdate>
			<TwitterDialog handleTwitterDialogClose={handleTwitterDialogClose}
										 twitterDialogOpen={twitterDialogOpen}
										 setTweetURL={setTweetURL}></TwitterDialog>
			<ImageUploadDialog imageUploadModalOpen={imageUploadModalOpen}
												 handleImageUploadDialogClose={handleImageUploadDialogClose}
												 setImageUploadModalOpen={setImageUploadModalOpen}
												 updateTitle={updateTitle}
												 setUpdateTitle={setUpdateTitle}
												 handleChange={handleChange}
												 onImageDrop={onImageDrop}
												 time={time}></ImageUploadDialog>
			<PreviewDialog previewModalOpen={previewModalOpen}
										 handlePreviewModalClose={handlePreviewModalClose}
										 setPreviewModalOpen={setPreviewModalOpen}
										 handleChange={handleChange}
										 tweetID={tweetID}
										 time={time}></PreviewDialog>
			<Feed updates={updates}></Feed>
		</div>
	)
}
