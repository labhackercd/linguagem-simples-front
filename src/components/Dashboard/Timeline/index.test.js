import React from 'react';
import { screen } from '@testing-library/react';
import {shallow, render} from "enzyme";
import Timeline from './index';
import Header from './Header';
import StatusSelection from './StatusSelection';
import NewUpdate from './NewUpdate';
import Feed from './Feed';
import SummaryBox from './SummaryBox';

it("should render the timeline section", () => {
    const component = shallow(<Timeline/>);
    expect(component).toMatchSnapshot();
});
