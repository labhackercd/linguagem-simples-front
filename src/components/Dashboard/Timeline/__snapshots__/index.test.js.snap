// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the timeline section 1`] = `
<Timeline
  classes={
    Object {
      "body": "Timeline-body-1",
    }
  }
/>
`;
